src/main/java/org/jsoup/select/NodeFilter.java

```diff
@@ -0,0 +1,82 @@
+package org.jsoup.select;
+
+import org.jsoup.nodes.Node;
+
+/**
+ * Node filter interface. Provide an implementing class to {@link NodeTraversor}
+ * to iterate through nodes.
+ * <p>
+ * This interface provides two methods, {@code head} and {@code tail}. The head
+ * method is called when the node is first seen, and the tail method when all of
+ * the node's children have been visited. As an example, head can be used to
+ * create a start tag for a node, and tail to create the end tag.
+ * </p>
+ * <p>
+ * For every node, the filter has to decide whether to
+ * <ul>
+ * <li>continue ({@link FilterResult#CONTINUE}),</li>
+ * <li>skip all children ({@link FilterResult#SKIP_CHILDREN}),</li>
+ * <li>skip node entirely ({@link FilterResult#SKIP_ENTIRELY}),</li>
+ * <li>remove the subtree ({@link FilterResult#REMOVE}),</li>
+ * <li>interrupt the iteration and return ({@link FilterResult#STOP}).</li>
+ * </ul>
+ * The difference between {@link FilterResult#SKIP_CHILDREN} and
+ * {@link FilterResult#SKIP_ENTIRELY} is that the first will invoke
+ * {@link NodeFilter#tail(Node, int)} on the node, while the latter does not.
+ * </p>
+ *
+ * @author Erich Schubert
+ */
+public interface NodeFilter {
+    /**
+     * Filter decision.
+     * 
+     * @author Erich Schubert
+     */
+    public enum FilterResult {
+	/** Continue processing the tree */
+	CONTINUE, //
+	/**
+	 * Skip the child nodes (for use in {@link NodeFilter#head(Node, int)}
+	 * only, equivalent to CONTINUE in {@link NodeFilter#tail(Node, int)}).
+	 * 
+	 * This will call {@link NodeFilter#tail(Node, int)} on the node next.
+	 */
+	SKIP_CHILDREN, //
+	/**
+	 * Skip the subtree, and do not call {@link NodeFilter#tail(Node, int)}.
+	 */
+	SKIP_ENTIRELY, //
+	/** Remove the node and its children */
+	REMOVE, //
+	/** Stop processing */
+	STOP //
+    }
+
+    /**
+     * Callback for when a node is first visited.
+     *
+     * @param node
+     *            the node being visited.
```

Comment:
Consistency: The rest of the code base puts the description on the same line as `@param`.

This issue also occurs in a couple of other places I don't individually point out.